### PR TITLE
Adding javadoc package-info.java files.

### PR DESCRIPTION
**Pull request by awholegunch**
Adds javadoc package-info.java files with descriptions to the tigon-api directory.
This also adds fixes for the Javadoc build, fixing some missing references.
It also replaces references to "Continuuity" and "Cask, Inc." with "Cask Data, Inc."

### CODE REVIEW

File: tigon-docs/developer-guide/source/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,440 @@
+.. :author: Cask Data, Inc.
+   :description: Index document
+   :copyright: Copyright © 2014 Cask Data, Inc.
+
+============================================
+Tigon Architecture
+============================================
+
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/architecture.rst]: JJ, I think the Architecture doc needs some change. Tigon consists of two parts - Big Flow (real-time stream processing framework implemented using Flows and Queues) and the second part is the DataStream Management System (aka GSTool). DSMS will look like a special Flowlet to the BigFlow system.

File: tigon-docs/developer-guide/source/developers.rst [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+There are multiple ways to connect the Flowlets of a Flow. The most
+common form is to use the Flowlet name. Because the name of each Flowlet
+defaults to its class name, when building the Flow specification you can
+simply write::
+
+  .withFlowlets()
+    .add(new RandomGenerator())
+    .add(new RoundingFlowlet())
+  .connect()
+    .fromStream("RandomGenerator").to("RoundingFlowlet")
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/developers.rst]: No Streams in Tigon. So a connection can be between only two Flowlets.

File: tigon-docs/developer-guide/source/developers.rst [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+  public static class WordCountFlow implements Flow {
+    @Override
+    public FlowSpecification configure() {
+      return FlowSpecification.Builder.with()
+        .setName("WordCountFlow")
+        .setDescription("Flow for counting words")
+        .withFlowlets().add(new Tokenizer())
+                       .add(new WordsFilter("the"))
+                       .add(new WordsCounter())
+        .connect().fromStream("text").to("Tokenizer")
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/developers.rst]: Here too. No Streams.

File: tigon-docs/developer-guide/source/developers.rst [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+
+Queues
+======
+
+    Check with Nitin
+
+Operations
+==========
+
+    Check with Nitin
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/developers.rst]: Operations:
Distribtued Mode :  will be the CLI commands info.
Standalone Mode : Only a single flow can be run and it is started when the run_standalone.sh script is invoked.

File: tigon-docs/developer-guide/source/index.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+============================================
+
+.. toctree::
+   :maxdepth: 2
+   
+   Introduction </intro>
+   Concepts </concepts>
+   Architecture </architecture>
+   Developers Guide </developers>
+   Administration </administration>
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/index.rst]: Most likely there will be no Admin guide

File: tigon-docs/developer-guide/source/index.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+   Troubleshooting </troubleshooting>
+   APIs </apis/apis>
```
Comment (gokulavasan) [inline tigon-docs/developer-guide/source/index.rst]: Should we have Javadocs API link separately. That is probably what devs might be interested in (at least in the beginning).

**Comment by awholegunch** [comment]
I have removed all the developer guide files that were added in error in this branch; they are in one of the other PRs. This PR is now just the javadocs changes. Please review again and comment.

File: tigon-api/src/main/java/co/cask/tigon/api/annotation/package-info.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * Annotations used when creating Applications.
```
Comment (gokulavasan) [inline tigon-api/src/main/java/co/cask/tigon/api/annotation/package-info.java]: when creating Flows

File: tigon-api/src/main/java/co/cask/tigon/api/common/package-info.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+/**
+ * Commonly used utility functions.
+ * <p>
+ * {@link co.cask.tigon.api.common.Bytes} class handles byte arrays, conversions to/from other types,
+ * comparisons, hash code generation, and manufacturing keys for HashMaps or HashSets.
+ */
+package co.cask.tigon.api.common;
+
```
Comment (gokulavasan) [inline tigon-api/src/main/java/co/cask/tigon/api/common/package-info.java]: Extra lines

File: tigon-api/src/main/java/co/cask/tigon/api/flow/flowlet/package-info.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * Interfaces used in Flows and Flowlets.
+ */
+package co.cask.tigon.api.flow.flowlet;
+
```
Comment (gokulavasan) [inline tigon-api/src/main/java/co/cask/tigon/api/flow/flowlet/package-info.java]: Extra lines

**Comment by gokulavasan** [comment]
Few comments. Other than that some of the package-info files don't have docs/comments above the package statement. Not sure if it is required.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Pull request merged**